### PR TITLE
Implement `_get_childCount` for UIA to avoid infinite recursion inn some cases

### LINKED ISSUE #11444: Python 3 versions of NVDA: NVDA crashes with recursion error when accessing childCount in initOverlayClass in some cases
This has been originally noticed because [Columns Review](https://github.com/ABuffEr/columnsReview) by @ABuffer  was causing crashes for users.

### Steps to reproduce:
1. Place [attached global plugin](https://github.com/nvaccess/nvda/files/5006814/testing.py.txt) into globalPlugins in the scratchpad removing the .txt extension.

There are two scenarios in which crashes occurs:
#### Scenario 1 in Windows 10 task manager:
- Ensure that view  in the task manager is set to "more details"
- Open task manager with ctrl + shift + escape and try to navigate in the list of processes.
#### Scenario 2 In Windows 7 list of networks:
- Ensure that you have more than one network available e.g VPN tunnel in addition to your main connection etc.
- Move to the system tray with Win + b
- Navigate to the list of networks, open it with space and try to navigate between available connections with arrow keys.

### Actual behavior:
In both scenarios NVDA crashes.
Log from the first scenario:

ERROR - NVDAObjects.__call__ (13:29:26.288) - Dummy-2 (8520):
Exception in chooseNVDAObjectOverlayClasses for <'taskmgr' (appName 'taskmgr', process ID 2972) at address 56772d0>
Traceback (most recent call last):
  File "NVDAObjects\__init__.pyc", line 95, in __call__
  File "appModules\taskmgr.pyc", line 31, in chooseNVDAObjectOverlayClasses
  File "baseObject.pyc", line 41, in __get__
  File "baseObject.pyc", line 144, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 1304, in _get_presentationType
  File "baseObject.pyc", line 41, in __get__
  File "baseObject.pyc", line 144, in _getPropertyViaCache
  File "NVDAObjects\__init__.pyc", line 744, in _get_presentationType
  File "baseObject.pyc", line 41, in __get__
  File "baseObject.pyc", line 144, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 1238, in _get_states
  File "NVDAObjects\UIA\__init__.pyc", line 835, in _prefetchUIACacheForPropertyIDs
  File "comtypesMonkeyPatches.pyc", line 26, in __call__
ctypes.ArgumentError: argument 1: <class 'RecursionError'>: maximum recursion depth exceeded in __instancecheck__

Log from the second scenario:

ERROR - NVDAObjects.__call__ (13:46:31.275) - Dummy-3 (5556):
Exception in event_NVDAObject_init for <'explorer' (appName 'explorer', process ID 3752) at address 45ecc30>
Traceback (most recent call last):
  File "NVDAObjects\__init__.pyc", line 153, in __call__
  File "appModules\explorer.pyc", line 368, in event_NVDAObject_init
  File "baseObject.pyc", line 41, in __get__
  File "baseObject.pyc", line 144, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 1331, in _get_parent
  File "NVDAObjects\__init__.pyc", line 86, in __call__
  File "NVDAObjects\UIA\__init__.pyc", line 852, in findOverlayClasses
  File "baseObject.pyc", line 41, in __get__
  File "baseObject.pyc", line 144, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 1115, in _get_TextInfo
  File "baseObject.pyc", line 41, in __get__
  File "baseObject.pyc", line 144, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 1100, in _get_UIATextPattern
  File "NVDAObjects\UIA\__init__.pyc", line 1079, in _getUIAPattern
  File "comtypesMonkeyPatches.pyc", line 26, in __call__
RecursionError: maximum recursion depth exceeded while calling a Python object

### Expected behavior:
No crash.
#### NVDA installed/portable/running from source:
tested with both installed and portable - no difference
#### NVDA version:
Crashes occurs with every version since Python 3 migration - first noticed with 2019.3 and still reproducible with 2022 and latest Alpha.
#### Windows version:
Windows 10 2004 for the first scenario Windows 7 X64 for the second.
#### Name and version of other software in use when reproducing the issue:
None
#### Other information about your system:
If the line `c = self.childCount` in the attached globalPlugin is commented out crash does not occur.
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
With 2019.2.1.  and earlier there is no crash with the same plugin so this is a regression.
#### If addons are disabled, is your problem still occuring?
No but this is expected -  the report is about NVDA interaction with the globalPlugin.
#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
Yes - no difference

### PR DESCRIPTION
**Pull request by lukaszgo1**
### Link to issue number:
Fixes #11444
### Summary of the issue:
When accessing `childCount` of UIA object in `initOverlayClass` in some cases NVDA crashed due to recursion error. While it isn't entirely clear why the error occurs in the first place @michaelDCurran proposed to implement a getter for `childCount` which instead of initializing objects for every child simply returns length of all cached UIA children.
### Description of how this pull request fixes the issue:
The property has been implemented as described above.
### Testing performed:
With global plugin from #11444 tested navigating in Windows 10 task manager and in list of networks in Windows 7 - previously NVDA crashed in both places whereas now it does not.
### Known issues with pull request:
While the crash is fixed we don't know what causes the recursion error in the first place.
### Change log entry:
I don't think any is needed - this was happening only with one add-on and its author worked around it disabling it for the affected controls. I'll certainly follow with the developer of Columns Review to no longer do that after this is merged.

### CODE REVIEW

**Comment by josephsl** [comment]
Forget about the comment I made in #11444 - I'll take a look at it today.

Thanks.

**Review by josephsl — APPROVED** [review]
Hi,

Looks fine - I may need to test this branch more to figure out issues if any. Thanks.

**Comment by josephsl** [comment]
Mick and Jamie, can I ask for a second look? Thanks.

File: source/NVDAObjects/UIA/__init__.py [hunk 1]
```diff
@@ -1404,23 +1404,39 @@ def _get_lastChild(self):
...
+
+	def _get_childCount(self):
+		try:
+			cachedChildren = self.UIAChildren
+			if not cachedChildren:
+				# GetCachedChildren returns null if there are no children.
+				return 0
+			return cachedChildren.length
+		except COMError:
+			return len(super().children)
```
Comment (leonardder) [inline source/NVDAObjects/UIA/__init__.py:1439]: Should this be:

```suggestion
			return super().childCount
```

**Comment by leonardder** [comment]
Forget my comment as well. super().childCount calls len(self.children), not len(super().children). That's a pitty. It would help if we could tell super to treat self as being super when doing these calls.

**Review by michaelDCurran — APPROVED** [review]

**Pull request merged**